components/HttpsCoponent_/HttpsRequest_.c

```diff
@@ -0,0 +1,315 @@
+        }
+        else if (ret != ESP_TLS_ERR_SSL_WANT_READ && ret != ESP_TLS_ERR_SSL_WANT_WRITE)
+        {
+            ESP_LOGE(TAG, "esp_tls_conn_write  returned: [0x%02X](%s)", ret, esp_err_to_name(ret));
+            goto cleanup;
+        }
+    } while (written_bytes < strlen(REQUEST));
+    ESP_LOGI(TAG, "Reading HTTP response...");
+    do
+    {
+        len = sizeof(buf) - 1;
+        memset(buf, 0x00, sizeof(buf));
+        ret = esp_tls_conn_read(tls, (char *)buf, len);
+
+        if (ret == ESP_TLS_ERR_SSL_WANT_WRITE || ret == ESP_TLS_ERR_SSL_WANT_READ)
+        {
+            continue;
+        }
+        else if (ret < 0)
+        {
+            ESP_LOGE(TAG, "esp_tls_conn_read  returned [-0x%02X](%s)", -ret, esp_err_to_name(ret));
+            break;
+        }
+        else if (ret == 0)
+        {
+            ESP_LOGI(TAG, "connection closed");
+            vTaskDelete(xTaskHandlerHTTPS);
+            break;
+        }
+
+        len = ret;
+        ESP_LOGD(TAG, "%d bytes read", len);
+        buf[len + 1] = '\n';
+        printf("\n\n\n%s\n\n\n", buf);
+        xSemaphoreGive(GetResponseSemaphore);
+        if (xQueueSend(BufQueue1, buf, 0) == pdTRUE)
+        {
+            printf("Sent data with queue \n");
+        }
+    } while (1);
+
+cleanup:
+    esp_tls_conn_destroy(tls);
+exit:
+    for (int countdown = 10; countdown >= 0; countdown--)
+    {
+        ESP_LOGI(TAG, "%d...", countdown);
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+}
+
+#if CONFIG_MBEDTLS_CERTIFICATE_BUNDLE
+static void https_get_request_using_crt_bundle(void)
+{
+    ESP_LOGI(TAG, "https_request using crt bundle");
+    esp_tls_cfg_t cfg = {
+        .crt_bundle_attach = esp_crt_bundle_attach,
+    };
+    https_get_request(cfg, Web_URL, HttpsBuf);
+}
+#endif // CONFIG_MBEDTLS_CERTIFICATE_BUNDLE
+
+static void https_get_request_using_cacert_buf(void)
+{
+    ESP_LOGI(TAG, "https_request using cacert_buf");
+    esp_tls_cfg_t cfg = {
+        .cacert_buf = (const unsigned char *)server_root_cert_pem_start,
+        .cacert_bytes = server_root_cert_pem_end - server_root_cert_pem_start,
+    };
+    https_get_request(cfg, Web_URL, HttpsBuf);
+}
+
+static void https_get_request_using_global_ca_store(void)
+{
+    esp_err_t esp_ret = ESP_FAIL;
+    ESP_LOGI(TAG, "https_request using global ca_store");
+    esp_ret = esp_tls_set_global_ca_store(server_root_cert_pem_start, server_root_cert_pem_end - server_root_cert_pem_start);
+    if (esp_ret != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Error in setting the global ca store: [%02X] (%s),could not complete the https_request using global_ca_store", esp_ret, esp_err_to_name(esp_ret));
+        return;
+    }
+    esp_tls_cfg_t cfg = {
+        .use_global_ca_store = true,
+    };
+    https_get_request(cfg, Web_URL, HttpsBuf);
+    esp_tls_free_global_ca_store();
+}
+
+#ifdef CONFIG_EXAMPLE_CLIENT_SESSION_TICKETS
+static void https_get_request_to_local_server(const char *url)
```

Comment:
is this function to "send request to local server" to "get request from local server"?
anyway, the name is too long,

Reply:
if I add some comment that show what function exactly do is better , I will do it

Reply:
the name needs to be clear also